transfer_test.go

```diff
@@ -0,0 +1,161 @@
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, t)
+
+	// Execute transfer-files
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.RtRepo1+";"+tests.RtRepo2))
+
+	// Verify files were transferred to the target Artifactory
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, targetServerDetails, t)
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, targetServerDetails, t)
+}
+
+func TestTransferExcludeRepo(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	// Upload files
+	runRt(t, "upload", "testdata/a/*", tests.RtRepo1)
+	runRt(t, "upload", "testdata/a/b/*", tests.RtRepo2)
+
+	// Verify files were uploaded to the source Artifactory
+	repo1Spec, err := tests.CreateSpec(tests.SearchAllRepo1)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, t)
+	repo2Spec, err := tests.CreateSpec(tests.SearchRepo2)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, t)
+
+	// Execute transfer-files
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.RtRepo1+";"+tests.RtRepo2, "--exclude-repos="+tests.RtRepo2))
+
+	// Verify repo1 files were transferred to the target Artifactory
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, targetServerDetails, t)
+	inttestutils.VerifyExistInTargetArtifactory([]string{}, repo2Spec, targetServerDetails, t)
+}
+
+func TestTransferMaven(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	assert.NoError(t, runMaven(t, createSimpleMavenProject, tests.MavenConfig, "install"))
+
+	// Verify files were uploaded to the source Artifactory
+	mvnRepoSpec, err := tests.CreateSpec(tests.SearchAllMaven)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetMavenDeployedArtifacts(), mvnRepoSpec, t)
+
+	// Execute transfer-files
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.MvnRepo1))
+
+	// Verify maven files were transferred to the target Artifactory
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetMavenDeployedArtifacts(), mvnRepoSpec, targetServerDetails, t)
+
+	// Wait for creation of maven-metadata.xml in the target Artifactory
+	inttestutils.WaitForCreationInArtifactory(tests.MvnRepo1+"/org/jfrog/cli-test/maven-metadata.xml", targetServerDetails, t)
+}
+
+func TestTransferPaginationAndThreads(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	// Decrease the AQL pagination limit to 10
+	transferfiles.AqlPaginationLimit = 10
+	defer func() {
+		transferfiles.AqlPaginationLimit = transferfiles.DefaultAqlPaginationLimit
+	}()
+
+	// Upload 100 files to the source Artifactory
+	for i := 0; i < 100; i++ {
```

Comment:
You may want to change this number to a number that is divided with the limit with a remainder (101)